### PR TITLE
Use "visible" scope when finding traces

### PR DESCRIPTION
**Pull request by AntonKhorev**
Allows to eliminate some :not_found branches.

### CODE REVIEW

File: app/controllers/traces_controller.rb [hunk 1]
```diff
@@ -68,10 +68,9 @@ def index
   end
 
   def show
-    @trace = Trace.find(params[:id])
+    @trace = Trace.visible.find(params[:id])
 
-    if @trace&.visible? &&
-       (@trace&.public? || @trace&.user == current_user)
+    if @trace&.public? || @trace&.user == current_user
```
Comment (tomhughes) [inline app/controllers/traces_controller.rb:73]: I realise it's not a change from this PR but I don't think we need the conditional dereference here do we? The `find` will have raised an exception if nothing was found...

**Review by tomhughes — COMMENTED** [review]
Also this could be extended to the icon and picture actions, or rather to the show actions in the new controllers. Those need to check `inserted` as well which could either be done with a where or with a new scope on trace.

File: app/controllers/traces_controller.rb (hunk 1 above)
Comment (AntonKhorev) [reply app/controllers/traces_controller.rb]: replaced `&.` with `.` here

**Comment by AntonKhorev** [comment]
Added `Trace.visible.find` to icon/picture controllers. Later I want to move this visibility check to CanCanCan ability conditions.

**Comment by tomhughes** [comment]
Looks good now, thanks.

**Pull request merged**